charts/ingress-nginx/templates/controller-deployment.yaml

```diff
@@ -29,11 +29,14 @@ spec:
   minReadySeconds: {{ .Values.controller.minReadySeconds }}
   template:
     metadata:
-    {{- if .Values.controller.podAnnotations }}
+    {{- if or .Values.controller.podAnnotations .Values.controller.podAnnotationConfigChecksum }}
       annotations: 
       {{- range $key, $value := .Values.controller.podAnnotations }}
         {{ $key }}: {{ $value | quote }}
       {{- end }}
+      {{- if .Values.controller.podAnnotationConfigChecksum }}
+        checksum/config: {{ tpl (toYaml .Values.controller) . | sha256sum }}
```

Comment:
why are you checksum'ing the whole of the values under the key `controller` this isn't the configmap.

I think we auto-reload the configuration on changes, which out make this unnecessary. Correct me if I'm wrong @aledbf

Reply:
I am not sure what is `checksum/config` being used here.

Reply:
`checksum/config` is only being used to trigger a Rollout Restart when a controller configuration is changed (`controller.*` from the chart values). I just wanted to keep the PR as similar as possible from original approved one.